File: c#/_97_LargeNonMersennePrime.cs
using System;
using System.Numerics;

public class _97_LargeNonMersennePrime{
	static long mod = 10_000_000_000;
	
	public static long Euler97(){
		// alternatively, use BigInteger.ModPow(base, exp, mod)
		BigInteger res = Pow(2, 7_830_457);
		res = (28433 * res + 1) % mod;
		return (long) res;
	}
	
	public static BigInteger Pow(BigInteger x, int n){
		BigInteger res = 1;
		while (n > 0){
			if (n % 2 == 1){
				res = (x * res) % mod;
			}
			x = (x * x) % mod;
			n /= 2;
		}
		return res;
	}
	
	public static void Main(){
		Console.WriteLine(Euler97());
	}
}

File: java/_97_LargeNonMersennePrime.java
import java.util.*;
import java.math.BigInteger;

public class _97_LargeNonMersennePrime{
    
    // private static long 
    
    public static void main(String[] args){
        // 28433 * (2 ^ 7830457) + 1
        BigInteger nmp = BigInteger.valueOf(2)
            .pow(7830457)
            .multiply(BigInteger.valueOf(28433))
            .mod(BigInteger.valueOf(10_000_000_000L))
            .add(BigInteger.ONE);
        
        System.out.println("Last 10 digits of given non-Mersenne prime: " + nmp); // 
    }
}
